framework/src/main/java/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -243,7 +243,17 @@ public AnnotatedTypeMirror visitDeclared(DeclaredType t, Void unused) {
                 }
             }
 
-            annotatedDeclaredType.setTypeArguments(typeArgs);
+            List<AnnotatedTypeMirror> typeArgReplacements = new ArrayList<>(typeArgs.size());
+            for (int i = 0; i < typeArgs.size(); i++) {
+                AnnotatedTypeMirror typeArg = typeArgs.get(i);
+                if (typeArg.getKind() == TypeKind.WILDCARD) {
```

Comment:
Couldn't all this be done within the code above?
If the underlying type is raw, I guess the wildcard is initialized properly.
In the non-raw case, can you simply change the `typeArgs.add(visit(javaTypeArg));` to do the check for a wildcard there?
Or change the behavior of `visit` and make sure that a wildcard is initialized properly?

Reply:
We'll investigate this later, together with #1089.